src/pseudopeople/interface.py

```diff
@@ -38,53 +53,88 @@ def _generate_dataset(
-    if not data_paths:
-        raise DataSourceError(
-            f"No datasets found at directory {str(source)}. "
-            "Please provide the path to the unmodified root data directory."
+        source = pathlib.Path(source)
+
+    if engine == "pandas":
+        # We process shards serially
+        data_paths = fetch_filepaths(dataset, source)
+        if not data_paths:
+            raise DataSourceError(
+                f"No datasets found at directory {str(source)}. "
+                "Please provide the path to the unmodified root data directory."
+            )
+
+        validate_data_path_suffix(data_paths)
+
+        # Iterate sequentially
+        noised_dataset = []
+        iterator = (
+            tqdm(data_paths, desc="Noising data", leave=False)
+            if len(data_paths) > 1
+            else data_paths
         )
-    validate_data_path_suffix(data_paths)
-    noised_dataset = []
-    iterator = (
-        tqdm(data_paths, desc="Noising data", leave=False)
-        if len(data_paths) > 1
-        else data_paths
-    )
 
-    for data_path in iterator:
-        logger.debug(f"Loading data from {data_path}.")
-        data = _load_data_from_path(data_path, user_filters)
-        if data.empty:
-            continue
-        data = _reformat_dates_for_noising(data, dataset)
-        data = _coerce_dtypes(data, dataset)
-        noised_data = noise_dataset(dataset, data, configuration_tree, seed)
-        noised_data = _extract_columns(dataset.columns, noised_data)
-        noised_dataset.append(noised_data)
+        for data_path in iterator:
+            logger.debug(f"Loading data from {data_path}.")
+            data = _load_data_from_path(data_path, user_filters, engine=engine)
+            if len(data.index) == 0:
+                continue
+            noised_data = _prep_and_noise_dataset(data, dataset, configuration_tree, seed)
+            noised_dataset.append(noised_data)
 
-    noised_dataset = pd.concat(noised_dataset, ignore_index=True)
+        noised_dataset = pandas.concat(noised_dataset, ignore_index=True)
 
-    # Known pandas bug: pd.concat does not preserve category dtypes so we coerce
-    # again after concat (https://github.com/pandas-dev/pandas/issues/51362)
-    noised_dataset = _coerce_dtypes(noised_dataset, dataset)
+        # Known pandas bug: pandas.concat does not preserve category dtypes so we coerce
+        # again after concat (https://github.com/pandas-dev/pandas/issues/51362)
+        noised_dataset = _coerce_dtypes(noised_dataset, dataset)
+    else:
+        import modin.pandas as mpd
+
+        # Let modin deal with how to partition the shards -- just one data path
+        data_path = source / dataset.name
+        data = _load_data_from_path(data_path, user_filters, engine=engine)
+
+        # HACK: This is using private Modin APIs
+        # How do we do this using the public API?
+        # See https://github.com/modin-project/modin/issues/6498
```

Comment:
Nit: HACK doesn't get highlighted by IDEs, so I'd suggest replacing this with FIXME, which does.
```suggestion
        # FIXME: This is using private Modin APIs
        #  How do we do this using the public API?
        #  See https://github.com/modin-project/modin/issues/6498
```